packages/backend/src/panels/AbstractWebviewPanel.ts

```diff
@@ -105,6 +108,24 @@ export abstract class AbstractWebviewPanel {
 
   protected dispose() {
     this.setFocused(false);
+    const isGeneratorCompleted = get(this.webViewPanel, Constants.GENERATOR_COMPLETED);
+    const yeomanui: any = get(this, "yeomanui");
+    // Verify the dispose happened before the user has finished the wizard and not by clicking "Finish".
+    // When user clicks "Finish" the generated has ended and the success/failure result will be set in "GENERATOR_COMPLETED".
+    // If it has not been ended, "GENERATOR_COMPLETED" will be undefined.
+    if (yeomanui && isGeneratorCompleted === undefined) {
+      const promptItems: any = get(yeomanui, "gen.prompts.items") ?? [];
+      const currentPromptCount = yeomanui.promptCount;
+      const numOfPromopts = promptItems.length;
+      const wizardStepName = promptItems[currentPromptCount - 1].name;
+      AnalyticsWrapper.updateGeneratorClosedManually(
+        yeomanui.generatorName,
```

Comment:
```suggestion
        yeomanui.generatorName ?? "",
```
a race condition may occur here - this property will be cleared after the generator has finished running.

Reply:
done